### PR TITLE
Add IOR benchmark package

### PR DESCRIPTION
**Pull request by adamjstewart**
@tgamblin: Do you happen to know who at LLNL develops IOR? If so maybe you can point them at this package and make sure everything looks good. I'm assuming `--with-hdf` means `hdf5+mpi` and `--with-ncmpi` means `netcdf+mpi` but I'm not really sure.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ior/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    def install(self, spec, prefix):
+        os.system('./bootstrap')
+
+        config_args = [
+            'MPICC=%s' % spec['mpi'].prefix.bin + '/mpicc',
+            '--prefix=%s' % prefix,
+        ]
+
+        if '+hdf5' in spec:
+            config_args.append('--with-hdf5')
```
Comment (mathstuf) [inline var/spack/repos/builtin/packages/ior/package.py]: Should it explicitly disable hdf5 in an else branch?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ior/package.py]: I don't think that's necessary. It's off by default.

Comment (mathstuf) [inline var/spack/repos/builtin/packages/ior/package.py]: Just making sure. Not all packages are that way :) .

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ior/package.py]: Yeah I know, and we don't really have a good standard convention either way. Lots of packages enable/disable things that are already on/off by default. Hopefully fewer packages forget to disable them...

Comment (tgamblin) [inline var/spack/repos/builtin/packages/ior/package.py]: I think it's generally nice to be explicit -- it doesn't hurt, and if the package changes its default the spack package doesn't break...

File: var/spack/repos/builtin/packages/ior/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from spack import *
+import os
+
+class Ior(Package):
+    """The IOR software is used for benchmarking parallel file systems
+    using POSIX, MPI-IO, or HDF5 interfaces."""
+
+    homepage = "http://www.nersc.gov/users/computational-systems/cori/nersc-8-procurement/trinity-nersc-8-rfp/nersc-8-trinity-benchmarks/ior/"
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/ior/package.py]: I'd use the IOR github page for the homepage.

**Comment by tgamblin** [comment]
@adamjstewart: I think I know who the POC is but lemme check.

**Comment by rhedges1** [comment]
There is no one at LLNL that is developing IOR, since ~2008ish.  The netcdf and hdf5 options are as you suspect (i.e. with MPI), but largely have been abandoned.

**Comment by adamjstewart** [comment]
If that's the case then I think this PR is ready to be merged.

**Pull request merged**